### DISCUSSION TITLE
Documentation

### DISCUSSION THREAD

**Discussion by ftomassetti** [TypeFox/typir#19] [General]
Is there any documentation or overview about this component?

**Comment by JohannesMeierSE** [comment]
Not yet, sorry 🙁 

We are still in early development of Typir, but we will provide an overview and a roadmap for Typir soon.

BTW I just enabled [GH Discussions](https://github.com/TypeFox/typir/discussions) now 🙂
Feel free to ask and share ideas!

We will present Typir at two conferences soon in order to discuss with the community about type checking in the web:
- [LangDev'24](https://langdevcon.org/2024/program#26)
- [OCX/EclipseCon'24](https://www.ocxconf.org/event/2024/websitePage:26bc9d91-446a-49e1-9f08-8fcc4680073e?session=23b97df9-0435-4fab-8a01-e0a9cf3e3831&shareLink=true)

For a first glimpse, have a look into an application of Typir for OX (a reduced version of LOX) with Langium [here](https://github.com/TypeFox/typir/blob/main/examples/ox/src/language/ox-type-checking.ts).

**Comment by koenvanexem** [comment]
Are you recording any of those talks?

**Comment by ftomassetti** [comment]
@KoenVanExem the talks at LangDev'24 should be recorded

**Reply by JohannesMeierSE** [reply]
Here are the slides, the recording will follow: https://github.com/TypeFox/typir/blob/main/resources/talks/2024-10-17-LangDev.pdf

**Comment by JohannesMeierSE** [comment]
@KoenVanExem the EclipseCon'24 will be recorded as well and published on Youtube

**Reply by JohannesMeierSE** [reply]
Here are [video](https://www.youtube.com/watch?v=WLzXAhcl-aY&list=PLy7t4z5SYNaRRGVdF83feN-_uHLwvGvgw&index=23) and [slides](https://github.com/TypeFox/typir/blob/main/resources/talks/2024-10-24-EclipseCon.pdf)

**Comment by JohannesMeierSE** [comment]
Here is the LangDev'24 recording of the Typir presentation: https://www.youtube.com/watch?v=CL8EbJYeyTE
